pydantic/_internal/_decorators.py

```diff
@@ -310,48 +333,51 @@ def gather_decorator_functions(cls: type[Any]) -> DecoratorInfos:
 
     # reminder: dicts are ordered and replacement does not alter the order
     res = DecoratorInfos()
-    for base in cls.__bases__:
+    for base in cls.__bases__[::-1]:
         existing = cast(Union[DecoratorInfos, None], getattr(base, '__pydantic_decorators__', None))
         if existing is not None:
-            res.validator.update(existing.validator)
-            res.field_validator.update(existing.field_validator)
-            res.root_validator.update(existing.root_validator)
-            res.field_serializer.update(existing.field_serializer)
-            res.model_serializer.update(existing.model_serializer)
+            res.validator.update({k: v.bind_to_cls(cls) for k, v in existing.validator.items()})
+            res.field_validator.update({k: v.bind_to_cls(cls) for k, v in existing.field_validator.items()})
+            res.root_validator.update({k: v.bind_to_cls(cls) for k, v in existing.root_validator.items()})
+            res.field_serializer.update({k: v.bind_to_cls(cls) for k, v in existing.field_serializer.items()})
+            res.model_serializer.update({k: v.bind_to_cls(cls) for k, v in existing.model_serializer.items()})
```

Comment:
This is the key here. It ensures we get the value from the child instance if it gets overriden